03week/towersOfHanoi.js

```diff
@@ -19,24 +39,62 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+function reset(){
+  stacks = {
+    a: [4, 3, 2, 1],
+    b: [],
+    c: []
+  };
+}
 
+function gameOver(){
+  printStacks();
+  reset();
 }
 
-function isLegal() {
-  // Your code here
+function movePiece(startStack, endStack) {
+  //Your code here
+  //endStack.push(startStack.pop());
+  const pickedNumber = startStack.pop();
+  endStack.push(pickedNumber);
+} 
 
+function isLegal(startStack, endStack) {
+  // Your code here
+  if (startStack[startStack.length - 1] < endStack[endStack.length - 1] || endStack.length == 0) {
+    return true;
+  } else {
+    return false;
+  }
 }
 
 function checkForWin() {
   // Your code here
-
+  if (stacks.c.length == 4) {
+    console.log("Victory!");
+    gameOver();
```

Comment:
gaveOver should be called in the parent function, not here, the purpose of this function is just to check for the win